src/nvim/mouse.c

```diff
@@ -303,6 +304,73 @@ int jump_to_mouse(int flags,
     mouse_past_bottom = true;
   }
 
+  // Adjust the clicked column position if there are concealed characters
+  // before the current column.  But only when it's absolutely necessary.
```

Comment:
I was thinking this, but I was trying to avoid adding new functions since this is my first stab at Neovim's source.  What are your thoughts on having a `syn_is_concealed()` or something that returns the match ID if the column is concealed?  It felt bad repeating the check.